lib/queuePopulator/IngestionProducer.js

```diff
@@ -0,0 +1,210 @@
+const async = require('async');
+const bucketclient = require('bucketclient');
+const Logger = require('werelogs').Logger;
+const MetadataWrapper = require('arsenal').storage.metadata.MetadataWrapper;
+
+// const config = require('../../conf/Config');
+const constants = require('../../constants');
+const RaftLogEntry = require('../../extensions/utils/RaftLogEntry');
+
+class IngestionProducer {
+    /**
+     * Create an IngestionProducer class that helps create a snapshot of
```

Comment:
This presentation of the class can actually go above the `class IngestionProducer` declaration as class JSDoc.